sdk/iot/hub/src/az_iot_hub_client_sas.c

```diff
@@ -15,109 +15,146 @@
 #define AMPERSAND '&'
 #define EQUAL_SIGN '='
 #define SCOPE_DEVICES_STRING "/devices/"
+#define SCOPE_MODULES_STRING "/modules/"
 #define SAS_TOKEN_SR "SharedAccessSignature sr"
 #define SAS_TOKEN_SE "se"
 #define SAS_TOKEN_SIG "sig"
 #define SAS_TOKEN_SKN "skn"
 
-az_result az_iot_sas_token_get_document(
-    az_span iothub_fqdn,
-    az_span device_id,
-    int32_t expiry_time_secs,
-    az_span document,
-    az_span* out_document)
+static const az_span devices_string = AZ_SPAN_LITERAL_FROM_STR(SCOPE_DEVICES_STRING);
+static const az_span modules_string = AZ_SPAN_LITERAL_FROM_STR(SCOPE_MODULES_STRING);
+static const az_span skn_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SKN);
+static const az_span sr_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SR);
+static const az_span sig_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SIG);
+static const az_span se_string = AZ_SPAN_LITERAL_FROM_STR(SAS_TOKEN_SE);
+
+AZ_NODISCARD az_result az_iot_hub_client_sas_get_signature(
+    az_iot_hub_client const* client,
+    uint32_t token_expiration_epoch_time,
+    az_span signature,
```

Comment:
pardon my ignorance here but should this API have the same dynamic as the following ones with `char*` and `length` as output? Or is this because the result here gets funneled straight into the following API's?

Reply:
no worries about questions, feel free to ask anything.
That's a good point, I'll clarify with the team.

Reply:
@danewalton - signature is not a string (null terminated) and so should be kept in a span (ptr, len). Password is a string though and that's the reason it changed into char*, size_t.

Reply:
gotcha thanks for the clarification